Arch Linux build failure: Declaration issues

**Describe the bug**

When attempting to compile ags on Arch Linux, I'm met with a number of errors regarding `*printf()` functions not being defined:

```
ags-v.3.5.0.24/Engine/ac/global_debug.cpp: In function ‘void script_debug(int, int)’:
ags-v.3.5.0.24/Engine/ac/global_debug.cpp:133:13: error: ‘sprintf’ was not declared in this scope
  133 |             sprintf(inroomtex, "!Enter new room: (in room %d)", displayed_room);
      |             ^~~~~~~
ags-v.3.5.0.24/Engine/ac/global_debug.cpp:40:1: note: ‘sprintf’ is defined in header ‘<cstdio>’; did you forget to ‘#include <cstdio>’?
   39 | #include "gfx/graphicsdriver.h"
  +++ |+#include <cstdio>
   40 | #include "main/graphics_mode.h"
make[2]: *** [Engine/CMakeFiles/engine.dir/build.make:746: Engine/CMakeFiles/engine.dir/ac/global_debug.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
ags-v.3.5.0.24/Engine/ac/global_display.cpp: In function ‘void Display(const char*, ...)’:
ags-v.3.5.0.24/Engine/ac/global_display.cpp:47:5: error: ‘vsprintf’ was not declared in this scope; did you mean ‘usprintf’?
   47 |     vsprintf(displbuf, get_translation(texx), ap);
      |     ^~~~~~~~
      |     usprintf
ags-v.3.5.0.24/Engine/ac/global_display.cpp: In function ‘void DisplayTopBar(int, int, int, const char*, const char*)’:
ags-v.3.5.0.24/Engine/ac/global_display.cpp:61:5: error: ‘snprintf’ was not declared in this scope
   61 |     snprintf(topBar.text, sizeof(topBar.text), "%s", get_translation(title));
      |     ^~~~~~~~
ags-v.3.5.0.24/Engine/ac/global_display.cpp:34:1: note: ‘snprintf’ is defined in header ‘<cstdio>’; did you forget to ‘#include <cstdio>’?
   33 | #include "main/game_run.h"
  +++ |+#include <cstdio>
   34 |
make[2]: *** [Engine/CMakeFiles/engine.dir/build.make:772: Engine/CMakeFiles/engine.dir/ac/global_display.cpp.o] Error 1
ags-v.3.5.0.24/Engine/ac/global_drawingsurface.cpp: In function ‘void RawPrintMessageWrapped(int, int, int, int, int)’:
ags-v.3.5.0.24/Engine/ac/global_drawingsurface.cpp:148:59: warning: declaration of ‘font’ shadows a global declaration [-Wshadow]
  148 | void RawPrintMessageWrapped (int xx, int yy, int wid, int font, int msgm) {
      |                                                       ~~~~^~~~
In file included from ags-v.3.5.0.24/build/_deps/allegro_content-src/include/allegro/base.h:41,
                 from ags-v.3.5.0.24/build/_deps/allegro_content-src/include/allegro.h:25,
                 from ags-v.3.5.0.24/Common/./util/wgt2allg.h:28,
                 from ags-v.3.5.0.24/Engine/./ac/draw.h:25,
                 from ags-v.3.5.0.24/Engine/ac/global_drawingsurface.cpp:17:
ags-v.3.5.0.24/build/_deps/allegro_content-src/include/allegro/text.h:34:23: note: shadowed declaration is here
   34 | AL_VAR(struct FONT *, font);
      |                       ^~~~
ags-v.3.5.0.24/build/_deps/allegro_content-src/include/allegro/internal/alconfig.h:217:64: note: in definition of macro ‘AL_VAR’
  217 |    #define AL_VAR(type, name)                      extern type name
      |                                                                ^~~~
make[2]: Leaving directory 'ags-v.3.5.0.24/build'
make[1]: *** [CMakeFiles/Makefile2:893: Engine/CMakeFiles/engine.dir/all] Error 2
make[1]: Leaving directory 'ags-v.3.5.0.24/build'
make: *** [Makefile:104: all] Error 2
make: Leaving directory 'ags-v.3.5.0.24/build'
```

Manually adding an `#include <stdio.h> in "Common/ac/common.h" fixes this issue, but then I'm met with some more:

```
ags-v.3.5.0.24/Engine/debug/filebasedagsdebugger.cpp:31:11: error: ‘::remove’ has not been declared; did you mean ‘uremove’?
   31 |         ::remove(SENT_MESSAGE_FILE_NAME);
      |           ^~~~~~
      |           uremove
ags-v.3.5.0.24/Engine/debug/filebasedagsdebugger.cpp:73:7: error: ‘::remove’ has not been declared; did you mean ‘uremove’?
   73 |     ::remove("dbgsend.tmp");
      |       ^~~~~~
      |       uremove

```

**AGS Version**
3.5.0.24

**To Reproduce**
Steps to reproduce the behavior:
1. Download latest source tarball release for linux
2. Compile with `cmake -DCMAKE_BUILD_TYPE=Release -B build && make -C build`

--- a/Common/font/fonts.cpp
+++ b/Common/font/fonts.cpp
@@ -12,6 +12,7 @@
 //
 //=============================================================================
 
+#include <cstdio>
 #include <vector>
 #include <alfont.h>
 #include "ac/common.h" // set_our_eip

--- a/Common/game/main_game_file.cpp
+++ b/Common/game/main_game_file.cpp
@@ -12,6 +12,7 @@
 //
 //=============================================================================
 
+#include <cstdio>
 #include "ac/audiocliptype.h"
 #include "ac/dialogtopic.h"
 #include "ac/gamesetupstruct.h"

--- a/Engine/ac/global_debug.h
+++ b/Engine/ac/global_debug.h
@@ -18,6 +18,7 @@
 #ifndef __AGS_EE_AC__GLOBALDEBUG_H
 #define __AGS_EE_AC__GLOBALDEBUG_H
 
+#include <cstdio>
 #include "util/string.h"
 
 AGS::Common::String GetRuntimeInfo();

--- a/Engine/ac/global_display.cpp
+++ b/Engine/ac/global_display.cpp
@@ -12,6 +12,7 @@
 //
 //=============================================================================
 
+#include <cstdio>
 #include <stdarg.h>
 #include "ac/common.h"
 #include "ac/character.h"

--- a/Engine/ac/gui.cpp
+++ b/Engine/ac/gui.cpp
@@ -12,6 +12,7 @@
 //
 //=============================================================================
 
+#include <cstdio>
 #include "ac/gui.h"
 #include "ac/common.h"
 #include "ac/draw.h"

--- a/Engine/ac/parser.cpp
+++ b/Engine/ac/parser.cpp
@@ -13,6 +13,7 @@
 //=============================================================================
 
 #include <cctype> //isalnum()
+#include <cstdio>
 #include "ac/common.h"
 #include "ac/gamesetupstruct.h"
 #include "ac/gamestate.h"

--- a/Engine/ac/string.cpp
+++ b/Engine/ac/string.cpp
@@ -12,6 +12,7 @@
 //
 //=============================================================================
 
+#include <cstdio>
 #include "ac/string.h"
 #include "ac/common.h"
 #include "ac/display.h"

--- a/Engine/ac/translation.cpp
+++ b/Engine/ac/translation.cpp
@@ -12,6 +12,7 @@
 //
 //=============================================================================
 
+#include <cstdio>
 #include "ac/asset_helper.h"
 #include "ac/common.h"
 #include "ac/gamesetup.h"

--- a/Engine/debug/filebasedagsdebugger.cpp
+++ b/Engine/debug/filebasedagsdebugger.cpp
@@ -12,6 +12,7 @@
 //
 //=============================================================================
 
+#include <cstdio>
 #include "debug/filebasedagsdebugger.h"
 #include "ac/file.h"                    // filelength()
 #include "util/stream.h"

--- a/Engine/gui/guidialog.cpp
+++ b/Engine/gui/guidialog.cpp
@@ -12,6 +12,7 @@
 //
 //=============================================================================
 
+#include <cstdio>
 #include "gui/guidialog.h"
 
 #include "ac/common.h"

--- a/Engine/script/cc_instance.cpp
+++ b/Engine/script/cc_instance.cpp
@@ -12,6 +12,7 @@
 //
 //=============================================================================
 
+#include <cstdio>
 #include <string.h>
 #include "ac/common.h"
 #include "ac/dynobj/cc_dynamicarray.h"

